### PR TITLE
fTPM vs dTPM

### PR DESCRIPTION
**Pull request by BartekLew**

### CODE REVIEW

**Comment by BartekLew** [comment]
@pietrushnic @mscibisz   please review

File: post.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Secure Boot, another official firmware without fTPM (if exists) could be used.
+If there is no Secure Boot, virtually any firmware could be used. Similarly, we
+would analyze the OS security level in the case of sTPM.
+
+The much more tricky part is particular functions provided by TPM. For example,
+how secure storage is implemented? Perhaps it's in SPI chip? sTPM would probably
+need to use regular files, but if we have Intel SGX it could provide secure
+storage. On different platforms, we have different security facilities, the most
+popular are TXT and SGX for Intel and TrustZone for ARM. There could be other
+ones too.
```
Comment (pietrushnic) [inline post.md]: @BartekLew above 2 paragraphs have very little facts. I think we will leave this blog post as is and when I will find time to give my 2 cents I will fill that gap.

**Comment by pietrushnic** [comment]
@miczyg1  at some point we have to get back to that. IMO it is can be very interesting blog post, but diving in datasheets may be required I can ask internally if there is any spec that explains things better.

**Comment by miczyg1** [comment]
@pietrushnic what I would do:

1. Intro to TPMs, what is it. Also division into fTPM, dTPM and sTPM is welcome, as already initiated in this PR.
2. TPM intro may be a preceding to TPM supported disk enryption, another thing we have on roadmap.
3. We once thought about fTPM in on apu2. The PSP offers firmware based TPM 2.0. It could be another blog post.

Many datasheets are available from TCG regarding dTPMs. I wonder about Intel related things. About AMD solutions we have documentation (PSP, AGESA etc.). Diving is unavoidable either way.

**Comment by pietrushnic** [comment]
@miczyg1 in some internal communication I get information that key consideration for fTPM should be:
1. If it provide implmentation of shielded locations 
2. If it provide implementation of protected objects
with high degree of confidence.

Also I get information that to brand legit fTPM you have to pass TPM conformance test suite which is available only to TCG members.

I'm looking where TCG define fTPM and requirements for it.

**Comment by pietrushnic** [comment]
@miczyg1 some input to this blog post:
* 12.2.5: https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-1-Architecture-01.38.pdf
* 4.3.3: https://trustedcomputinggroup.org/wp-content/uploads/TCG_1_4_Architecture_Overview.pdf
* 9.3: https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-1-Architecture-01.38.pdf

**Comment by pietrushnic** [comment]
@miczyg1 let me know if you have plans and time to finish this blog post.

**Comment by miczyg1** [comment]
@pietrushnic I have plans and I will willingly finish the post. Would You like to cooperate?

**Comment by pietrushnic** [comment]
@miczyg1 does @mkopec publication close this MR?

**Comment by miczyg1** [comment]
I believe it should. Closing

**Pull request closed without merging**